modin/data_management/functions/default_methods/groupby_default.py

```diff
@@ -153,18 +185,33 @@ def fn(
             axis=0,
             is_multi_by=None,
             drop=False,
+            selection=None,
             **kwargs
         ):
             """Group DataFrame and apply aggregation function to each group."""
             by = cls.validate_by(by)
 
             grp = df.groupby(by, axis=axis, **groupby_args)
+
+            if selection is not None:
+                grp = grp[selection]
+
             agg_func = cls.get_func(grp, key, **kwargs)
             result = (
                 grp.agg(agg_func, **agg_args)
                 if isinstance(agg_func, dict)
                 else agg_func(grp, **agg_args)
             )
+            if (
```

Comment:
this condition has the same meaning as [here](https://github.com/modin-project/modin/pull/3298/files#r687855805), the difference is that the base backend already handles all of the cases naturally except this one with the `selection`. I

Ideally, we want to move all of the exceptions checks to the front-end, this would help us to get rid of the code duplication for different backend in terms of raising proper errors and would help to get errors as soon as possible. But this is a task for a different PR